### ISSUE TITLE
Feature: Improve Values object to allow access to nested array values

### ISSUE THREAD

**Issue by proggeler** [auraphp/Aura.Web#58]
If a key within the Values container is an array I would like to access these array elements with the get* methods.

Current usage:
(assumes `<input type="checkbox" name="user[allowed]">` is checked and posted)
```
$user = $request->post->get('user');
$is_allowed = $user['allowed'] == 'on'; // $is_allowed = true
```
I would suggest to use a dot notation (like javascript) to gain access to the stored values:
```
// posted: <input type="checkbox" name="user[allowed]">
$is_allowed = $request->post->getBool('user.allowed'); // $is_allowed = true
```

Another way of gaining this way of access is ensure an array within the post values is also stored as a Values object, but this is backwards breaking.
```
// posted: <input type="checkbox" name="user[allowed]">
$is_allowed = $request->post->get('user')->getBool('allowed'); // $is_allowed = true
```

**Comment by harikt** [comment]
One Question : What happens if the currrent name used is having dot .

Eg : `<input type="checkbox" name="user.allowed">` 

`$is_allowed = $request->post->get('user.allowed');`

Will it break current returned values ?

**Comment by proggeler** [comment]
@harikt I've added a similar case to the unit tests. A name containing a dot will take precedence over a nested one. This way the change will be backwards compatible.

**Comment by harikt** [comment]
@proggeler cool. 

Thank you. I will look into it.

**Comment by harikt** [comment]
I have merged #59 . So looks this issue have been resolved.
